src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -987,18 +987,28 @@ private static string EnsurePackageExtension(string packagePath, bool isSnupkg)
         }
 
         /// <summary>
-        /// This method checks of a given list of package paths exist on disk.
+        /// If there isn't at least one Path specified, throw that no file paths were resolved for this Package.
         /// </summary>
-        /// <param name="packagePath">A package path to be used while creating logs.</param>
-        /// <param name="matchingPackagePaths">A list of matching package paths that need to be checked.</param>
+        /// <param name="packagePath">The package path the user originally provided for use while creating logs.</param>
+        /// <param name="matchingPackagePaths">A list of matching package paths that were previously resolved.</param>
         public static void EnsurePackageFileExists(string packagePath, IEnumerable<string> matchingPackagePaths)
         {
-            if (!matchingPackagePaths.Any())
+            if (!ResolvedAnyPackagePath(matchingPackagePaths))
             {
+                ErrorFileNotFound(packagePath);
+            }
+        }
+
+        public static void ErrorFileNotFound(string packagePath)
+        {
                 throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,
                     Strings.UnableToFindFile,
                     packagePath));
-            }
+        }
+
+        public static bool ResolvedAnyPackagePath(IEnumerable<string> matchingPackagePaths)
```

Comment:
```suggestion
        public static bool IsNotNullOrEmpty(IEnumerable<string> strings)
```

although this could be made an extension method for all `IEnumerable<T>`.

Reply:
Prefer a name that's more about intent, not a repeat of what the code does. Would like to discuss some of these principles and decide on a standard.

Reply:
This really does feel like a side thing without any "relevant" business logic as @zivkan mentioned in some of his earlier comments.

Reply:
Ok, my takeaway here is to remove this method, and copy the throw logic to the 2-3 places it's needed.